### ISSUE TITLE
the error about training model

### ISSUE THREAD

**Issue by ssm0808** [google/deepvariant#69]
I currently tried to write some script to train the new model. The make_example script have created labeled data. But the model_train script have some problem. I made some labeled examples using the WGS case study data of Deepvariant on chr20. I used the 0.6.0 version of released DeepVariant model as a started training model.

**The make_examples script is:**
`python ../bin/make_examples.zip \
  --mode training \
  --ref "file/ucsc.hg19.chr20.unittest.fasta.gz" \
  --reads "file/NA12878_S1.chr20.10_10p1mb.bam" \
  --confident_regions "file/test_nist.b37_chr20_100kbp_at_10mb.bed" \
  --truth_variants "file/test_nist.b37_chr20_100kbp_at_10mb.vcf.gz" \
  --examples "output/examples.tfrecord.gz"
`
**my-training-dataset.pbtxt file:**
`name: "my-training-dataset"
tfrecord_path: "/home/suanfa/Documents/wangpeng/testmake_examples/output/examples.tfrecord.gz"
num_examples: 1`

**The model_train script is:**
`python ../bin/model_train.zip \
  --dataset_config_pbtxt "./my-training-dataset.pbtxt" \
  --start_from_checkpoint "/my/path/of/DeepVariant/deepvariant-model-wes_and_wgs/DeepVariant-inception_v3-0.6.0+cl-191676894.data-wgs_standard/model.ckpt"
`

**The following error have happened while the model_train.zip is invoked:**
> I0502 10:58:51.903573 139632719935232 model_train.py:182] Initializing model from checkpoint at /home/suanfa/Documents/source/DeepVariant/deepvariant-model-wes_and_wgs/DeepVariant-inception_v3-0.6.0+cl-191676894.data-wgs_standard/model.ckpt
2018-05-02 10:58:56.347500: I tensorflow/core/platform/cpu_feature_guard.cc:137] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 AVX512F FMA
2018-05-02 10:58:57.263635: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1030] Found device 0 with properties:
name: Tesla P100-PCIE-12GB major: 6 minor: 0 memoryClockRate(GHz): 1.3285
pciBusID: 0000:3b:00.0
totalMemory: 11.91GiB freeMemory: 11.62GiB
2018-05-02 10:58:57.263682: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1120] Creating TensorFlow device (/device:GPU:0) -> (device: 0, name: Tesla P100-PCIE-12GB, pci bus id: 0000:3b:00.0, compute capability: 6.0)
INFO:tensorflow:Restoring parameters from /tmp/deepvariant/model.ckpt-0
I0502 10:58:57.455770 139632719935232 tf_logging.py:82] Restoring parameters from /tmp/deepvariant/model.ckpt-0
INFO:tensorflow:Starting Session.
I0502 10:59:09.842276 139632719935232 tf_logging.py:82] Starting Session.
INFO:tensorflow:Saving checkpoint to path /tmp/deepvariant/model.ckpt
I0502 10:59:10.099534 139621333726976 tf_logging.py:82] Saving checkpoint to path /tmp/deepvariant/model.ckpt
INFO:tensorflow:Starting Queues.
I0502 10:59:10.102293 139632719935232 tf_logging.py:82] Starting Queues.
INFO:tensorflow:global_step/sec: 0
I0502 10:59:13.668776 139621325334272 tf_logging.py:121] global_step/sec: 0
INFO:tensorflow:Recording summary at step 0.
I0502 10:59:14.875045 139621316941568 tf_logging.py:82] Recording summary at step 0.
INFO:tensorflow:global step 1: loss = 0.2608 (4.963 sec/step)
I0502 10:59:15.326091 139632719935232 tf_logging.py:82] global step 1: loss = 0.2608 (4.963 sec/step)
2018-05-02 10:59:15.584978: E tensorflow/core/kernels/check_numerics_op.cc:157] abnormal_detected_host @0x104ef6dce00 = {1, 0} LossTensor is inf or nan
2018-05-02 10:59:15.615399: W tensorflow/core/framework/op_kernel.cc:1192] Invalid argument: LossTensor is inf or nan : Tensor had NaN values
         [[Node: train_op/CheckNumerics = CheckNumerics[T=DT_FLOAT, message="LossTensor is inf or nan", _device="/job:localhost/replica:0/task:0/device:GPU:0"](control_dependency_4)]]
INFO:tensorflow:Error reported to Coordinator: <class 'tensorflow.python.framework.errors_impl.InvalidArgumentError'>, LossTensor is inf or nan : Tensor had NaN values
         [[Node: train_op/CheckNumerics = CheckNumerics[T=DT_FLOAT, message="LossTensor is inf or nan", _device="/job:localhost/replica:0/task:0/device:GPU:0"](control_dependency_4)]]
         [[Node: train_op/control_dependency/_5647 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device_incarnation=1, tensor_name="edge_14228_train_op/control_dependency", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:CPU:0"]()]]
 Caused by op u'train_op/CheckNumerics', defined at:
  File "/tmp/Bazel.runfiles_ecWAzH/runfiles/com_google_deepvariant/deepvariant/model_train.py", line 364, in <module>
    tf.app.run()
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/platform/app.py", line 48, in run
    _sys.exit(main(_sys.argv[:1] + flags_passthrough))
  File "/tmp/Bazel.runfiles_ecWAzH/runfiles/com_google_deepvariant/deepvariant/model_train.py", line 352, in main
    parse_and_run()
  File "/tmp/Bazel.runfiles_ecWAzH/runfiles/com_google_deepvariant/deepvariant/model_train.py", line 314, in parse_and_run

**Comment by pichuan** [comment]
The content of the file seems wrong to me:

```
name: "my-training-dataset"
tfrecord_path: "/home/suanfa/Documents/wangpeng/testmake_examples/output/examples.tfrecord.gz" num_examples: 1
```

Specifically, do you really just have one example in that `examples.tfrecord.gz` file? It doesn't really make sense to train on a file with one record, and I actually suspect weird things could happen.

Can you create a more realistic training file, say, with at least 10,000 examples, and try again?

**Comment by ssm0808** [comment]
We use the testdata from DeepVariant source (https://github.com/google/deepvariant/tree/r0.6/deepvariant/testdata) .The test_nist.b37_chr20_100kbp_at_10mb.bed is really small, so we just can set one example in that examples.tfrecord.gz. 

**test_nist.b37_chr20_100kbp_at_10mb.bed file:**
> chr20	10000846	10002407
chr20	10002520	10004171
chr20	10004273	10004964
chr20	10004994	10006386
chr20	10006409	10007800
chr20	10007824	10008018
chr20	10008043	10008079
chr20	10008100	10008707
chr20	10008808	10008897
chr20	10009002	10009791
chr20	10009933	10010531

We will try to train model with our realistic WES data, and set with at least 10,000 examples. Thanks

**Comment by ssm0808** [comment]
We have tried to train model with the realisitic WES data. The script can pass.We used GNU parallel tool to split the input and generates shared output. But we met a problem that we just can input one tfrecord file to the pbtxt file. How can we input all of the tfrecord file to train model at the same time?

**The make_examples script is:**
`
BASE="${HOME}/Documents/source"
BIN_DIR="${BASE}/bin"
MODELS_DIR="${BASE}/DeepVariant/deepvariant-model-wes_and_wgs/DeepVariant-inception_v3-0.6.0+cl-191676894.data-wgs_standard"
MODEL="${MODELS_DIR}/model.ckpt"
N_SHARDS="64"
BAM="/sdbdata/WESdata/CL100026859_bwa/CL100026859_L02_14/PE150.2.rmdup.bam"
REF="/home/suanfa/Documents/source/ref/hg19.fasta"
var=${BAM##*/}
var=${var%.*}
path="/home/suanfa/Documents/shishiming/training_WES_model"
OUTPUT_DIR="$path/output"
EXAMPLES="${OUTPUT_DIR}/${var}.training.examples.tfrecord@${N_SHARDS}.gz"
CONFIDENT_REGIONS="/home/suanfa/Documents/source/exome_region_bed/new_v4.bed"
TRUTH_VARIANTS="/home/suanfa/Documents/source/NISTv3.3_baseline/HG001_GRCh37_GIAB_highconf_CG-IllFB-IllGATKHC-Ion-10X-SOLID_CHROM1-X_v.3.3.2_highconf_PGandRTGphasetransfer.vcf.new_ch    rID.vcf.gz"
LOG_DIR="${OUTPUT_DIR}/logs" 

echo 'Run make_examples'
( time seq 0 $((N_SHARDS-1)) | \
  parallel --halt 2 --joblog "${LOG_DIR}/log" --res "${LOG_DIR}" \
      python "${BIN_DIR}"/make_examples.zip \
          --mode training \
          --ref "${REF}" \
         --reads "${BAM}" \
         --examples "${EXAMPLES}" \
         --confident_regions ${CONFIDENT_REGIONS} \
         --truth_variants ${TRUTH_VARIANTS} \
         --task {}
  ) >"${LOG_DIR}/make_examples.log" 2>&1

`

**the part of the examples output:**
> PE150.2.rmdup.training.examples.tfrecord-00000-of-00064.gz PE150.2.rmdup.training.examples.tfrecord-00022-of-00064.gz  PE150.2.rmdup.training.examples.tfrecord-00044-of-00064.gz
PE150.2.rmdup.training.examples.tfrecord-00001-of-00064.gz  PE150.2.rmdup.training.examples.tfrecord-00023-of-00064.gz  PE150.2.rmdup.training.examples.tfrecord-00045-of-00064.gz
PE150.2.rmdup.training.examples.tfrecord-00002-of-00064.gz  PE150.2.rmdup.training.examples.tfrecord-00024-of-00064.gz  PE150.2.rmdup.training.examples.tfrecord-00046-of-00064.gz
PE150.2.rmdup.training.examples.tfrecord-00003-of-00064.gz  PE150.2.rmdup.training.examples.tfrecord-00025-of-00064.gz  PE150.2.rmdup.training.examples.tfrecord-00047-of-00064.gz
PE150.2.rmdup.training.examples.tfrecord-00004-of-00064.gz  PE150.2.rmdup.training.examples.tfrecord-00026-of-00064.gz  PE150.2.rmdup.training.examples.tfrecord-00048-of-00064.gz
PE150.2.rmdup.training.examples.tfrecord-00005-of-00064.gz  PE150.2.rmdup.training.examples.tfrecord-00027-of-00064.gz  PE150.2.rmdup.training.examples.tfrecord-00049-of-00064.gz
PE150.2.rmdup.training.examples.tfrecord-00006-of-00064.gz  PE150.2.rmdup.training.examples.tfrecord-00028-of-00064.gz  PE150.2.rmdup.training.examples.tfrecord-00050-of-00064.gz
PE150.2.rmdup.training.examples.tfrecord-00007-of-00064.gz  
.....

Just input the first tfrecord file to my-training-dataset.pbtxt file
**my-training-dataset.pbtxt file:**
`
name: "my-training-dataset"
tfrecord_path: "/home/suanfa/Documents/shishiming/training_WES_model/output/PE150.2.rmdup.training.examples.tfrecord-00000-of-00064.gz"
num_examples: 64
`

**The model_train script is:**
`python /home/suanfa/Documents/source/bin/model_train.zip \
   --dataset_config_pbtxt "./my-training-dataset.pbtxt" \
   --start_from_checkpoint "/home/suanfa/Documents/source/DeepVariant/deepvariant-model-wes_and_wgs/DeepVariant-inception_v3-0.6.0+cl-191676894.data-wes_standard/model.ckpt" \
   --batch_size  32 \
   --train_dir "/tmp/deepvariant/wex" \
   --number_of_steps 30000 \
   --learning_rate 0.001
`

**Comment by pichuan** [comment]
Good to hear you're making progress.
You can use patterns like:
```
name: "train"
tfrecord_path: "/path/to/wes_train.examples-?????-of-00256.tfrecord.gz"
num_examples: 15705449
```

Note: in order for the training to work correctly, you need to shuffle the examples. Currently we rely on the examples already being shuffled, and we don't do shuffling in the tensorflow training parti.
